src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop,
+  State
+} from "@stencil/core";
+
+import { IComponent } from "../common/interfaces";
+
+@Component({
+  shadow: false,
+  styleUrl: "gauge.scss",
+  tag: "gx-gauge"
+})
+export class Gauge implements IComponent {
+  @Element() element: HTMLElement;
+
+  @Event() gxGaugeDidLoad: EventEmitter;
+
+  /**
+   * Property of type Style
+   * Define the color of the center in _(Circle)_ gauge type.
+   * Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   * _(Transparent by Default)_
+   */
+  @Prop() styleCenterColor = "transparent";
+
+  /**
+   * Property of type Style
+   * Define the color of the center text in _(Circle)_ gauge type.
+   * Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   * _(Gray by Default)_
+   */
+  @Prop() styleCenterTextColor = "gray";
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * Property of type Style.
+   * Define if border will display or not. Default is disabled.
+   */
+  @Prop() styleBorder = false;
+
+  /**
+   * Property of type Style.
+   * Define the border width. Value in *px*.
+   */
+  @Prop() styleBorderWidth = 2;
+
+  /**
+   * Property of type Style.
+   * Define the border color. Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   */
+  @Prop() styleBorderColor = "rgb(0, 92, 129)";
+
+  /**
+   * This property allows you to select the gauge type. _(Circle or Line)_.
+   * Default is linear type.
+   */
+  @Prop() gaugeType: "Line" | "Circle" = "Line";
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
```

Comment:
This property cannot be a CSS variable because it used within the code for calculations purposes.

Reply:
Still, I feel that, as it's used to calculate the value of other CSS properties, this is a good candidate for CSS variable, using the `calc()` function.

Reply:
😓 
Look for example the next line:
```typescript
(this.minimumSize <= 300 && this.thickness <= 25) || !this.styleShadow
   ? "none"
   : ""
```
It needs the thickness just right there, next to the `minimumSize` to evaluate if it active the shadows or not.
For example, if the user set shadows but the control is very little. There, it will evaluate if set shadows or not. It will depend on the